src/plot/highlight.js

```diff
@@ -0,0 +1,270 @@
+import React from 'react';
+import AbstractSeries from './series/abstract-series';
+import {getAttributeScale} from 'utils/scales-utils';
+import PropTypes from 'prop-types';
+
+function getLocs(evt) {
+  const xLoc = evt.type === 'touchstart' ? evt.pageX : evt.offsetX;
+  const yLoc = evt.type === 'touchstart' ? evt.pageY : evt.offsetY;
+  return {xLoc, yLoc};
+}
+
+class Highlight extends AbstractSeries {
+  state = {
+    dragging: false,
+    brushArea: {top: 0, right: 0, bottom: 0, left: 0},
+    brushing: false,
+    startLocX: 0,
+    startLocY: 0,
+    dragArea: null
+  };
+
+  _getDrawArea(xLoc, yLoc) {
+    const {startLocX, startLocY} = this.state;
+    const {allow, innerWidth, innerHeight, marginLeft, marginRight, marginBottom, marginTop} = this.props;
+    const plotHeight = innerHeight + marginTop + marginBottom;
+    const plotWidth = innerWidth + marginLeft + marginRight;
+
+    return {
+      bottom: allow.includes('y') ? Math.max(startLocY, yLoc) : plotHeight,
```

Comment:
I don't like the api of using an array of strings. Can this use booleans for horizontal and vertical or x and y?